apps/typegpu-docs/src/content/examples/rendering/3d-fish/tgsl-helpers.ts

```diff
@@ -12,3 +12,40 @@ export const distanceVectorFromLine = tgpu['~unstable']
     return std.sub(point, closestPoint);
   })
   .$name('distance vector from line');
+
+const ApplySinWaveReturnSchema = d.struct({
+  position: d.vec3f,
+  normal: d.vec3f,
+});
+
+export const applySinWave = tgpu['~unstable']
+  .fn([d.u32, d.u32, d.vec3f, d.vec3f], ApplySinWaveReturnSchema)
+  .does((index, time, position, normal) => {
+    const a = 60.1;
+    const b = 0.8;
+    const c = 10.1;
+    // z += sin(index + (time / a + x) / b) / c
+
+    const positionModification = d.vec3f(
+      0,
+      0,
+      std.sin(d.f32(index) + (d.f32(time) / a + position.x) / b) / c,
+    );
+
+    const coeff =
+      std.cos(d.f32(index) + (d.f32(time) / a + position.x) / b) / c;
+    const newOX = std.normalize(d.vec3f(1, 0, coeff));
+    const newOZ = d.vec3f(-coeff, 0, 1);
```

Comment:
I made a goof when patching this part of the code. Since `newOX` is normalized, it's not equivalent.
```suggestion
    const newOZ = d.vec3f(-newOX.z, 0, newOX.x);
```